core/src/idx/trees/mtree.rs

```diff
@@ -2688,4 +2686,10 @@ mod tests {
 		// Create a seeded RNG
 		StdRng::seed_from_u64(seed)
 	}
+
+	impl Vector {
+		fn clone_vector(&self) -> Vector {
```

Comment:
Why is this necessary? The `self.clone()` below suggests that it's already clonable. Why not just use that directly?

Reply:
That was only used by tests, cloning the inner Vector (and not the Arc<Vector>). Because the MTree::insert and MTree::delete methods were expecting a Vector and not a SharedVector. But that's not a requirement anymore, and I can update this methods to take a SharedVector. I am fixing that. Good point!

Reply:
> That was only used by tests, cloning the inner Vector (and not the Arc)

You could always dereference before cloning to get the inner Vector.